src/latexify/transformers/function_expander.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+import ast
+import functools
+from ast import Load
+from collections.abc import Callable
+
+from latexify import ast_utils
+
+
+# TODO(ZibingZhang): handle recursive function expansions
+class FunctionExpander(ast.NodeTransformer):
+    """NodeTransformer to expand functions.
+
+    This class replaces function calls with an expanded form.
+
+    Example:
+        def f(x, y):
+            return hypot(x, y)
+
+        FunctionExpander({"hypot"}) will modify the AST of the function above to below:
+
+        def f(x, y):
+            return sqrt(x**2, y**2)
+    """
+
+    def __init__(self, functions: set[str]) -> None:
+        self._functions = functions
+
+    def visit_Call(self, node: ast.Call) -> ast.AST:
+        """Visitor of Call nodes."""
+        func_name = ast_utils.extract_function_name_or_none(node)
+        if (
+            func_name is not None
+            and func_name in self._functions
+            and func_name in _FUNCTION_EXPANDERS
+        ):
+            return _FUNCTION_EXPANDERS[func_name](self, node)
+
+        return node
+
+
+def _hypot_expander(function_expander: FunctionExpander, node: ast.Call) -> ast.AST:
+    if len(node.args) == 0:
+        return ast_utils.make_constant(0)
+
+    args = [
+        ast.BinOp(function_expander.visit(arg), ast.Pow(), ast_utils.make_constant(2))
+        for arg in node.args
+    ]
+
+    args_reduced = functools.reduce(lambda a, b: ast.BinOp(a, ast.Add(), b), args)
+    return ast.Call(
+        func=ast.Name(id="sqrt", ctx=Load()),
```

Comment:
Maybe in `constants.py`, `BUILTIN_FUNCS` should be `dict[enum, tuple[str, str]]`? 

Basically what's happening here is that we're replacing `hypot` with `sqrt`, while also relying on the fact `sqrt` gets further preprocessed by getting replaced with `(r"\sqrt{", "}")`.

Having an enum rather than a bunch of strings might make the code a bit more readable, as well as helping to prevent errors with typos. In the future with more function expansions and different transformers, there might be a lot of cases where we care about specific, commonly-used functions, such as `sqrt`.

If you agree with changing it to an enum I'll tackle that in another issue / PR.

Reply:
To be clear, do you mean defining an Enum class something like:
```python
class BuiltinFnNames(str, enum.Enum):
    ...
    SQRT = "sqrt"
```

and use it in:
```python
BUILTIN_FUNCS: dict[BuiltinFnNames, tuple[str, str]] = {
    ...
    BuiltinFnNames.SQRT: (r"\sqrt{", "}"),
    ...
}
```
? If I'm wrong please fix me.

Reply:
something like that yeah

Reply:
It looks nicer than the current impl. I guess we finally need to adopt more sophisticated way to represent functions, but for now the idea above looks a good choice.